recipe-client-addon/lib/AddonStudies.jsm

```diff
@@ -112,6 +124,46 @@ this.AddonStudies = {
     };
   },
 
+  async init() {
+    // If the add-on an active study has been removed since we last ran, stop
+    // the study.
+    const activeStudies = (await this.getAll()).filter(study => study.active);
+    const db = await getDatabase();
+    for (const study of activeStudies) {
+      const addon = await AddonManager.getAddonByID(study.addonId);
+      if (!addon) {
+        this.markAsEnded(study);
+        await getStore(db).put(study);
+        Services.obs.notifyObservers(study, STUDY_ENDED_TOPIC);
```

Comment:
I hesitated doing that because I don't want to send the notification out until the save has actually happened. I didn't save inside the method because the add-on watcher needs to use its own database instance to save.

But I could pass in the database to use as a parameter to solve that. And it's pretty much an internal-only API anyway, I should just remove it from the exported object.